src/adapter/etl-adapter-excel/src/Flow/ETL/Adapter/Excel/ExcelExtractor.php

```diff
@@ -171,6 +170,8 @@ private function extractRows(SourceStream $stream, array $headers, int $offset)
                 $rows[] = $rowData;
             }
 
+            $this->reader()->close();
+
             return $rows;
```

Comment:
I just noticed that, there is a memory issue here. 
Instead of of mergning each $row into `$rows[] = $rowData` you should `yield `$rowData` (ideally rename it to $row). 

Extractors by design needs to yield one row at time, this behavior can be adjusted by the users through `batchSize()` or `collect()`